Userland/Utilities/mkfs.fat.cpp

```diff
@@ -0,0 +1,608 @@
+/*
+ * Copyright (c) 2024, the SerenityOS developers.
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <AK/Endian.h>
+#include <AK/Format.h>
+#include <AK/Types.h>
+#include <Kernel/FileSystem/FATFS/Definitions.h>
+#include <LibCore/ArgsParser.h>
+#include <LibCore/File.h>
+#include <LibCore/System.h>
+#include <LibFileSystem/FileSystem.h>
+#include <LibMain/Main.h>
+#include <sys/time.h>
+#include <unistd.h>
+
+// Public domain boot code adapted from:
+// https://github.com/dosfstools/dosfstools/blob/289a48b9cb5b3c589391d28aa2515c325c932c7a/src/mkfs.fat.c#L205
+
+// clang-format off
+static u8 s_bootcode[74] = {
```

Comment:
The use of raw encoded bytes of x86 instructions in this file means that it will only work on an x86_64 build of serenity, or on an x86_64 host. Should it not be guarded as such in the CMakeLists.txt?